### DISCUSSION TITLE
How are API keys generated?

### DISCUSSION THREAD

**Discussion by bson** [Part-DB/Part-DB-server#675] [General]
I can't seem to find anywhere to generate an API key for KiCad... checked under user settings, both as me and as admin, under the System tab... but nowhere do I find anything to generate an API key.  Where is it hiding?!

Running 1.13.2.  Is there anything I need to enable on the server side?

**Comment by jbtronics** [comment]
You probably need to give the API permission to your user, otherwise you will not be able to see the API token options on your user settings page.

**Comment by bson** [comment]
Thank you sir, indeed - the admin group didn't have API permissions by default.

Now as I populate the EDA fields of parts they show nicely.  Super sweet!  Only wish I could import all this in bulk from a file... Or rather, export all parts with incomplete EDA info, populate them with a script and/or manually in a spreadsheet, then reimport it.  A lot of the passives (I pretty much only use 0402/0603, anything else is less common and I can tell from the power or voltage rating) are trivial to fill in.

![Screenshot 2024-08-25 at 9 56 21 AM](https://github.com/user-attachments/assets/7be5afe3-2c43-455b-90ca-fa92f9d840f1)

**Comment by bson** [comment]
I do see a mistake I made... for parts like in the image that come in a multitude of makes, pinouts, and packages I should create a category, then have specific variations as parts under it.  Then in this case the part category (e.g. 74AHC08) can have the schematic symbol while the parts have the footprints.  Though in other cases it may be the opposite, for example FPGA grades - same pinout, same footprints, just different grade.

**Comment by bson** [comment]
Here's a suggestion: when sending parts data to KiCad, populate a custom field with available stock (total, all locations combined).
